src/lib/fetch.js

```diff
@@ -21,3 +34,13 @@ export function postJson(url, body, params = {}) {
   params.body = JSON.stringify(body);
   return fetchJson(url, params);
 }
+
+export function getData(options = {}) {
```

Comment:
This abstraction doesn't make things easier to follow. I think `getFilesData` can be interesting.

The switch between `getProfileData` and `getFilesData` can simply happen where it's needed.